addons/account/models/account_invoice.py

```diff
@@ -560,8 +560,20 @@ def _prepare_tax_line_vals(self, line, tax):
             'account_analytic_id': tax['analytic'] and line.account_analytic_id.id or False,
             'account_id': self.type in ('out_invoice', 'in_invoice') and (tax['account_id'] or line.account_id.id) or (tax['refund_account_id'] or line.account_id.id),
         }
+        # If the taxes generate moves on the same financial account as the invoice line,
+        # propagate the analytic account from the invoice line to the tax line.
+        # This is necessary in situations were (part of) the taxes cannot be reclaimed,
+        # to ensure the tax move is allocated to the proper analytic account.
+        if not vals['account_analytic_id'] and line.account_analytic_id and val['account_id'] == line.account_id.id:
+            val['account_analytic_id'] = line.account_analytic_id.id
         return vals
 
+    def _cleaned_tax_grouped(self, tax_grouped):
+        """
+        Hook method to manipulate invoice tax lines, based on advanced contitions.
+        """
+        return tax_grouped
```

Comment:
I absolutely agree, that technically it is more elegant. If you don't agree with the "political/hinting" aspects (https://github.com/odoo/odoo/pull/13998#discussion_r85304059), I think then cherry picking the [CLEANUP] commit would be the right thing to do at this point.